.pre-commit-config.yaml

```diff
@@ -43,7 +43,7 @@ repos:
       - id: clang-format
         name: clang-format
         description: Format files with ClangFormat.
-        entry: clang-format-10
+        entry: clang-format-12
```

Comment:
If you update, try to use `clang-format-14` as that is the default version on Ubuntu 22.04.

Reply:
Meaning [here](https://github.com/PickNikRobotics/rviz_visual_tools/blob/master/.github/workflows/format.yaml#L14) you can change the Ubuntu version to 22.04 for compatibility. That's likely why you ended up having to try out version 12.

Reply:
there is a problem in the test, it fails using clang 14.

Error: The process '/opt/hostedtoolcache/Python/3.11.3/x64/bin/pre-commit' failed with exit code 3

Reply:
right, because the format CI workflow is running in a Ubuntu 20.04 environment that's installing clang-format 10, according to [this file](https://github.com/PickNikRobotics/rviz_visual_tools/blob/master/.github/workflows/format.yaml#L14). So you may want to update that workflow as well and it should work!

Reply:
Oh sorry, this is is a ROS 1 branch so you actually want to stay in Ubuntu 20.04 / clang-format-10. Never mind

Reply:
You are right! unfortunately the master branch is on ros1.
I close the pull request.